### ISSUE TITLE
Stackoverflow errors when converting chain with large-dimensional parameter vector to idata object

### ISSUE THREAD

**Issue by mgmverburg** [arviz-devs/ArviZ.jl#140]
So I found that when using a relatively large dimensional parameter (specifically a 8x131, resulting in 1048 parameters for a single parameter group), I get a stackoverflow error resulting from trying to convert the chain to an idata object.

I found a way to reproduce it:
```julia
stackoverflowarray = [MCMCChains.AxisArray(rand(500,4)) for i in 1:1048]
cat(stackoverflowarray...; dims=3)
```
Which comes from this line in the library essentially:
https://github.com/arviz-devs/ArviZ.jl/blob/17f2e7e6957f262ffa04c0fd5155cb8952f4f5e2/src/mcmcchains.jl#L79
where `var_views` is similar to my `stackoverflowarray` above

In reality, it somehow seems caused by the AxisArray, because leave out the AxisArray part in my reproduction example and it doesn't happen.
Similarly, replacing the line:
https://github.com/arviz-devs/ArviZ.jl/blob/17f2e7e6957f262ffa04c0fd5155cb8952f4f5e2/src/mcmcchains.jl#L78
with
```julia
var_views = (Array(@view(chns.value[:, n, :])) for n in loc_names)
```
seems to fix it as well.

**Comment by mgmverburg** [comment]
Okay, I open an issue in AxisArrays!
I honestly did not know if the use case was appropriate or not, but in general it is quite odd that when using AxisArray and then doing cat(array..., dims=3) it consumes 6GB, vs not using an AxisArray consuming only 25MB and no errors.

https://github.com/JuliaArrays/AxisArrays.jl/issues/203

**Comment by sethaxen** [comment]
Given that AxisArrays has had no updates or interaction from the devs on the issues/PRs in ~6 months, it seems unlikely this will be handled there, so we should work around it here. A conversion as suggested in https://github.com/arviz-devs/ArviZ.jl/issues/140#issuecomment-905929587 should be sufficient.

**Comment by sethaxen** [comment]
Fixed by #156
